frontend/src/scenes/trends/trendsLogic.ts

```diff
@@ -512,7 +512,14 @@ export const trendsLogic = kea<
             actions.setBreakdownValuesLoading(false)
         },
         [eventDefinitionsLogic.actionTypes.loadEventDefinitionsSuccess]: async () => {
-            actions.setFilters(getDefaultFilters(values.filters, eventDefinitionsLogic.values.eventNames), true)
+            const newFilter = getDefaultFilters(values.filters, eventDefinitionsLogic.values.eventNames)
+            const mergedFilter: Partial<FilterType> = {
+                ...values.filters,
+                ...newFilter,
+            }
+            if (!objectsEqual(values.filters, mergedFilter)) {
```

Comment:
This is tangential, but I'm not 100% sure this `objectsEqual` works as expected. What if the filters are ordered differently?